NohanaImagePicker/MomentDetailListViewController.swift

```diff
@@ -0,0 +1,200 @@
+            let nohanaImagePickerController = nohanaImagePickerController,
+            let momentInfoSection = momentInfoSection else {
+                fatalError("failed to dequeueReusableCellWithIdentifier(\"AssetDetailCell\")")
+        }
+        cell.scrollView.zoomScale = 1
+        cell.tag = indexPath.item
+
+        let imageSize = CGSize(
+            width: cellSize.width * UIScreen.main.scale,
+            height: cellSize.height * UIScreen.main.scale
+        )
+        let asset = PhotoKitAsset(asset: momentInfoSection.assetResult[indexPath.item])
+        asset.image(targetSize: imageSize) { (imageData) -> Void in
+            DispatchQueue.main.async(execute: { () -> Void in
+                if let imageData = imageData {
+                    if cell.tag == indexPath.item {
+                        cell.imageView.image = imageData.image
+                        cell.imageViewHeightConstraint.constant = self.cellSize.height
+                        cell.imageViewWidthConstraint.constant = self.cellSize.width
+                    }
+                }
+            })
+        }
+        return (nohanaImagePickerController.delegate?.nohanaImagePicker?(nohanaImagePickerController, assetDetailListViewController: self, cell: cell, indexPath: indexPath, photoKitAsset: asset.originalAsset)) ?? cell
+    }
+
+    // MARK: - UICollectionViewDelegateFlowLayout
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return cellSize
+    }
+    
+    // MARK: - UIScrollViewDelegate
+
+    override func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        guard let collectionView = collectionView else {
+            return
+        }
+        let row = Int((collectionView.contentOffset.x + cellSize.width * 0.5) / cellSize.width)
+        if row < 0 {
+            currentIndexPath = IndexPath(row: 0, section: currentIndexPath.section)
+        } else if row >= collectionView.numberOfItems(inSection: 0) {
+            currentIndexPath = IndexPath(row: collectionView.numberOfItems(inSection: 0) - 1, section: currentIndexPath.section)
+        } else {
+            currentIndexPath = IndexPath(row: row, section: currentIndexPath.section)
+        }
+    }
+
+    // MARK: - IBAction
+    
+    @IBAction func didPushPickButton(_ sender: UIButton) {
+        guard let momentInfoSection = momentInfoSection else { return }
+        let asset = PhotoKitAsset(asset: momentInfoSection.assetResult[currentIndexPath.item])
+        if pickButton.isSelected {
+            if nohanaImagePickerController!.pickedAssetList.drop(asset: asset) {
+                pickButton.isSelected = false
+            }
+        } else {
+            if nohanaImagePickerController!.pickedAssetList.pick(asset: asset) {
+                pickButton.isSelected = true
+            }
+        }
```

Comment:
```suggestion
        guard let nohanaImagePickerController = nohanaImagePickerController,
              let momentInfoSection = momentInfoSection else { return }
        let asset = PhotoKitAsset(asset: momentInfoSection.assetResult[currentIndexPath.item])
        if pickButton.isSelected {
            if nohanaImagePickerController.pickedAssetList.drop(asset: asset) {
                pickButton.isSelected = false
            }
        } else {
            if nohanaImagePickerController.pickedAssetList.pick(asset: asset) {
                pickButton.isSelected = true
            }
        }
```
`Forced Unwrapping`が気になりました。